src/snowflake/cli/api/metrics.py

```diff
@@ -54,17 +67,125 @@ class CLICounterField:
     PACKAGE_SCRIPTS = f"{_TypePrefix.FEATURES}.{_DomainPrefix.APP}.package_scripts"
 
 
+class _CLIMetricsStep:
+    """
+    class for holding metrics step data and encapsulating related operations
+    """
+
+    _id_counter = count(start=1, step=1)
```

Comment:
since steps end up in a single snowflake table, the ID should probably be a UUID